src/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Read.HandleArray.cs

```diff
@@ -184,9 +192,7 @@ private static void HandleStartArray(JsonSerializerOptions options, ref ReadStac
                     JsonPropertyInfo jsonPropertyInfo = state.Current.JsonPropertyInfo;
 
                     object currentEnumerable = jsonPropertyInfo.GetValueAsObject(state.Current.ReturnValue);
-                    if (currentEnumerable == null ||
-                        // ImmutableArray<T> is a struct, so default value won't be null.
-                        jsonPropertyInfo.IsImmutableArray)
+                    if (currentEnumerable == null)
```

Comment:
`ImmutableArray<T>` will no longer come down this path.

Reply:
Because it's immutable, we use a converter then we [set the property directly](https://github.com/dotnet/corefx/blob/0bfa3eb975ac98ea82922a3b75e5ddf2e499a8fe/src/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Read.HandleArray.cs#L112)  as highlighted in https://github.com/dotnet/corefx/pull/42420/files#r345507426.